### PR TITLE
Thermal Safe Drill

### PR DESCRIPTION
**Pull request by Geevies**
* Station safes now spawn with items in them again.
* You can now buy a thermal drill capable of drilling into the station safe for 8 telecrystals. It will take five minutes to do so.

image

Ported from https://github.com/ParadiseSS13/Paradise/pull/10046

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Comment by sheeplets** [comment]
2112720434_preview_20200531_041545

**Comment by DickFreedomJohnson** [comment]
guys the thermal drill go get it

File: code/game/objects/structures/safe.dm [hunk 1]
```diff
@@ -21,24 +22,42 @@ FLOOR SAFES
 	var/space = 0		//the combined w_class of everything in the safe
 	var/maxspace = 24	//the maximum combined w_class of stuff in the safe
 
+	// Variables (Drill)
+	var/obj/item/thermal_drill/drill = null // The currently placed thermal drill, if any.
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/safe.dm:26]:
```suggestion
	var/obj/item/thermal_drill/drill // The currently placed thermal drill, if any.
```

File: code/game/objects/structures/safe.dm [hunk 2]
```diff
@@ -68,22 +87,64 @@ FLOOR SAFES
...
+				if(!drill || isprocessing)
+					return
+				if(broken)
+					to_chat(user, SPAN_WARNING("\The [src] is already broken open!"))
+					return
+				if(do_after(user, 2 SECONDS))
+					last_drill_time = world.time
+					drill.soundloop.start()
+					START_PROCESSING(SSprocessing, src)
+					update_icon()
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/safe.dm:120]: These loops need to check conditions again when the loop is complete. The drill might break or get removed in that timeframe.

File: code/game/objects/structures/safe.dm [hunk 3]
```diff
@@ -68,22 +87,64 @@ FLOOR SAFES
...
+			if("Turn Off")
+				if(!drill || !isprocessing)
+					return
+				if(do_after(user, 2 SECONDS))
+					drill.soundloop.stop()
+					STOP_PROCESSING(SSprocessing, src)
+					update_icon()
+			if("Remove Drill")
+				if(isprocessing)
+					to_chat(user, "<span class='warning'>You cannot remove the drill while it's running!</span>")
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/safe.dm:130]:
```suggestion
					to_chat(user, SPAN_WARNING("You cannot remove the drill while it's running!"))
```

**Review by NonQueueingMatt — DISMISSED** [review]

**Review by Alberyk — APPROVED** [review]

**Review by skull132 — APPROVED** [review]

**Pull request merged**